bittensor/_neuron/base_miner_neuron.py

```diff
@@ -0,0 +1,212 @@
+# The MIT License (MIT)
+# Copyright © 2023 Yuma Rao
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the “Software”), to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software,
+# and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+# the Software.
+
+# THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+# THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+
+import os
+import time
+import torch
+import threading
+import argparse
+import bittensor
+
+from rich import print
+from typing import Union, Tuple
+from datetime import datetime
+
+class BaseMinerNeuron:
```

Comment:
Similar question here; is this the base miner for the servers?

Reply:
We will have:
- TextToImage
- SpeechToText
- TextToCompletion (This is currently text_prompting and needs to extend this, working on it now)

So the main idea is to extend each multi-modal class from this `BaseMinerNeuron` that way we don't have entirely bespoke submodules.